### DISCUSSION TITLE
ClientApp on Win10 machine

### DISCUSSION THREAD

**Discussion by srobert72** [mattjlewis/diozero#144] [Q&A]
Hi,

I try to connect a client app on a Win10 machine.
But it fails on startup.
I seems that the **libdiozero-system-utils** library is missing for this kind of machine.
Is it possible to add this ?

Thx for your contribution

**Comment by EAGrahamJr** [comment]
Not that I can actually answer this, but out of curiosity, what hardware are you running?

**Reply by srobert72** [reply]
I'm using an Arduino Mega board.
I saw Matthew on a video having a connection directly between a laptop and an Arduino, without RaspberryPi.
I tried to do the same with my Win10 laptop, but it doesn't work.
I understand Matthew was surely using a MacOs ou Linux laptop.
I need a Win10 machine because I want to communicate with some flight simulation games.
I could deploy my Java app on a RaspberryPi but currently in heavy development phase it's easyer to debug directly on my laptop.

**Comment by mattjlewis** [comment]
Hopefully this will improve in the next release (1.4.0) where the native library will only be initialised when absolutely necessary.
Just need to find time to test latest features before releasing.
I have a Windows 10 Home edition PC as well as macOS laptop that I can test from.

**Reply by mattjlewis** [reply]
I’ll make sure it works in the next release

**Comment by mattjlewis** [comment]
I was a bit quick in promising support for this in the next release as this is bigger than just testing when the native library is loaded - I will need to implement native serial support for Windows for this to work with Arduino via serial (note that Arduino via WiFi should work already).
Not a huge undertaking as there is plenty of guidance out there. I try to keep the dependencies down as much as possible hence using my own serial implementation, rather than something like jSerialCom.

**Comment by mattjlewis** [comment]
Thinking more about this, I think I can fairly easily add an implementation that simply uses jSerialComm, just need to do some clever class loading (ideally service loader) to keep the jSerialComm code in a separate JAR file.

**Comment by mattjlewis** [comment]
Would definitely like to add support for my Firmata implementation on Windows.
